File: frontend2/Rugby/src/app/classes/associations/AthleteGameStats.ts
import {Game} from '../game';
import {Stats} from '../stats';
import {Athlete} from '../athlete';

export class AthleteGameStats {
  constructor(
    public id?: number,
    public athlete?: Athlete,
    public stats?: Stats,
    public game?: Game
  ) {
    this.id = id ? id : 0;
    this.athlete = athlete ? athlete : new Athlete();
    this.stats = stats ? stats : new Stats();
    this.game = game ? game : new Game();
  }
}

File: backend/src/main/java/pt/isel/rugby/model/AthleteGameStats.java
package pt.isel.rugby.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

import javax.persistence.*;

@Entity
@Table(name = "athleteGameStats")
@Data
public class AthleteGameStats {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToOne
    private Athlete athlete;

    @ManyToOne
    private Stats stats;

    @ManyToOne
    @JsonIgnore
    private Game game;
}
